test/microphysicsflexible_tests.jl

```diff
@@ -0,0 +1,83 @@
+import Test as TT
+
+import Thermodynamics as TD
+import CloudMicrophysics.Parameters as CMP
+import CloudMicrophysics.MicrophysicsFlexible as CMF
+import Cloudy.ParticleDistributions as CPD
+import Cloudy.KernelFunctions as CPK
+
+@info "Microphysics Tests"
+
+function test_microphysics_flexible(FT)
+    # Thermodynamics and air properties parameters
+    aps = CMP.AirProperties(FT)
+    tps = TD.Parameters.ThermodynamicsParameters(FT)
+
+    TT.@testset "Flexible microphysics - unit tests" begin
+        # Create Cloudy struct with different distributions 
+        clinfo = CMF.CLSetup{FT}(;
+            pdists = [
+                CPD.LognormalPrimitiveParticleDistribution(
+                    FT(10.0),
+                    FT(1.0),
+                    FT(1.0),
+                ),
+            ],
+            mom = CPD.get_moments(
+                CPD.LognormalPrimitiveParticleDistribution(
+                    FT(10.0),
+                    FT(1.0),
+                    FT(1.0),
+                ),
+            ),
+            NProgMoms = [3],
+            KernelFunc = CPK.ConstantKernelFunction(FT(1)),
+            mass_thresholds = [FT(100.0)],
+            kernel_order = 4,
+            kernel_limit = FT(100),
+        )
+
+        # Create Cloudy struct with defaults
+        clinfo = CMF.CLSetup{FT}()
+        TT.@test length(clinfo.pdists) == 2
+        TT.@test length(clinfo.mom) == 5
+        TT.@test length(clinfo.NProgMoms) == 2
+
+        # Test coalescence
```

Comment:
Should the mass be conserved in this approach when doing collisions? Could we test that?

Reply:
Yes, but these tests are already covered in Cloudy.jl so I think it would be redundant here

Reply:
These tests are only to check that the new features/structs that are created in CM.jl work. But, if you think it would be useful to create some redundant tests, I can absolutely do so! Let's open an issue if so.

Reply:
It's fine. Lets add as many tests as is necessary to test that Cloudy can be used in CM.